spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/PulsarPropertiesTests.java

```diff
@@ -335,12 +339,14 @@ void producerProperties() {
 					.containsEntry("batchingMaxMessages", 7).containsEntry("batchingMaxBytes", 8)
 					.containsEntry("batchingEnabled", false).containsEntry("chunkingEnabled", true)
 					.hasEntrySatisfying("encryptionKeys",
-							keys -> assertThat(((Set<String>) keys)).containsExactly("my-key"))
+							keys -> assertThat(keys).asInstanceOf(InstanceOfAssertFactories.collection(String.class))
```

Comment:
This would be a good candidate to create a helper method for this as its in N places. We can do that later though.